yarn-project/aztec/src/sandbox.ts

```diff
@@ -182,6 +183,26 @@ async function deployCanonicalL2GasToken(deployer: Wallet, l1ContractAddresses:
   logger.info(`Deployed Gas Token on L2 at ${canonicalGasToken.address}`);
 }
 
+/**
+ * Deploys the key registry on L2.
+ */
+async function deployCanonicalKeyRegistry(deployer: Wallet) {
+  const canonicalKeyRegistry = getCanonicalKeyRegistry();
+
+  if (await deployer.isContractClassPubliclyRegistered(canonicalKeyRegistry.contractClass.id)) {
+    return;
+  }
+
+  await new BatchCall(deployer, [
+    (await registerContractClass(deployer, canonicalKeyRegistry.artifact)).request(),
+    deployInstance(deployer, canonicalKeyRegistry.instance).request(),
+  ])
+    .send()
+    .wait();
+
```

Comment:
Please throw here an error if the address of the deployed contract does not match the constant.